modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php

```diff
@@ -0,0 +1,150 @@
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    // Return an empty array if empty, so the field can be considered empty.
+    if ($entity->get('oe_organisation_contact')->isEmpty()) {
+      return [];
+    }
+
+    $contact = $entity->get('oe_organisation_contact')->entity;
+    if (!$contact instanceof ContactInterface) {
+      return [];
+    }
+
+    $build = [
+      '#type' => 'pattern',
+      '#id' => 'field_list',
+      '#variant' => 'horizontal',
+      '#fields' => [
+        'items' => [],
+      ],
+    ];
+    $cache = CacheableMetadata::createFromRenderArray($build);
+
+    $contact_access = $contact->access('view', NULL, TRUE);
+    $cache->addCacheableDependency($contact_access);
+
+    if (!$contact_access->isAllowed()) {
+      $cache->applyTo($build);
+      return $build;
+    }
+
+    $cache->addCacheableDependency($contact);
+    $items = [];
+    $fields = [
+      'oe_website' => [],
+      'oe_email' => ['type' => 'email_mailto'],
+      'oe_phone' => [],
+      'oe_address' => [
+        'type' => 'oe_theme_helper_address_inline',
+        'settings' => ['delimiter' => ', '],
+      ],
+    ];
+    foreach ($fields as $field_name => $display_options) {
+      $item = $this->getRenderableFieldListItem($contact, $field_name, $display_options);
+      if (!empty($item)) {
+        $items[] = $item;
+      }
+    }
+    $build['#fields']['items'] = $items;
+    $cache->applyTo($build);
+
+    return $build;
+  }
+
+  /**
+   * Gets renderable item for field list pattern.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityInterface $entity
+   *   Content entity.
+   * @param string $field_name
+   *   Field name.
+   * @param array $display_options
+   *   Display options for field rendering.
+   *
+   * @return array
+   *   Renderable array.
+   */
+  protected function getRenderableFieldListItem(ContentEntityInterface $entity, string $field_name, array $display_options = []): array {
+    if ($entity->get($field_name)->isEmpty()) {
+      return [];
+    }
```

Comment:
Move this if before 109, so we only check for the empty field and we don't have to check for `if (!empty($item)) {`. It will be the responsibility of the called to call this function knowing that it will return.